docs/sources/flow/reference/components/pyroscope.scrape.md

```diff
@@ -32,45 +55,104 @@ pyroscope.scrape "LABEL" {
+`scheme`            | `string`                 | The URL scheme with which to fetch metrics from targets.     | `"http"`       | no
+`bearer_token`      | `secret`                 | Bearer token to authenticate with.                           |                | no
+`bearer_token_file` | `string`                 | File containing a bearer token to authenticate with.         |                | no
+`proxy_url`         | `string`                 | HTTP proxy to proxy requests through.                        |                | no
+`follow_redirects`  | `bool`                   | Whether redirects returned by the server should be followed. | `true`         | no
+`enable_http2`      | `bool`                   | Whether HTTP2 is supported for requests.                     | `true`         | no
+
+At most one of the following authentication mechanisms can be provided:
  - [`bearer_token` argument](#arguments).
  - [`bearer_token_file` argument](#arguments).
  - [`basic_auth` block][basic_auth].
  - [`authorization` block][authorization].
  - [`oauth2` block][oauth2].
 
- [arguments]: #arguments
+[arguments]: #arguments
+
+#### `job_name` argument
+
+`job_name` defaults to the component's unique identifier.
+
+For example, the `job_name` of `pyroscope.scrape "local" { ... }` will be `"pyroscope.scrape/local"`.
+
+#### `targets` argument
+
+The list of `targets` can be provided [statically][example_static_targets], [dynamically][example_dynamic_targets], 
+or a [combination of both][example_static_and_dynamic_targets].
+
+The special `__address__` label _must always_ be present and corresponds to the
+`<host>:<port>` that is used for the scrape request.
+
+Labels starting with a double underscore (`__`) are treated as _internal_, and are removed prior to scraping.
+
+The special label `service_name` is required and must always be present. 
+If it is not specified, `pyroscope.scrape` will attempt to infer it from 
+either of the following sources, in this order: 
+1. `__meta_kubernetes_pod_annotation_pyroscope_io_service_name` which is a `pyroscope.io/service_name` pod annotation.
+2. `__meta_kubernetes_namespace` and `__meta_kubernetes_pod_container_name`
+3. `__meta_docker_container_name`
+
+If `service_name` is not specified and could not be inferred, then it is set to `unspecified`.
+
+The following labels are automatically injected to the scraped profiles 
+so that they can be linked to a scrape target:
+
+| Label            | Description                                                      |
+|------------------|----------------------------------------------------------------- |
+| `"job"`          | The `job_name` that the target belongs to.                       |
+| `"instance"`     | The `__address__` or `<host>:<port>` of the scrape target's URL. |
+| `"service_name"` | The inferred Pyroscope service name.                             |
+
+#### `scrape_interval` argument
+
+If `scrape_interval` is short:
+* Advantages:
```

Comment:
One advantage is also being able to catch events such as OOM when they happen quickly. So maybe something like "Ability to capture rapid changes in profiles".

Reply:
I can see how this is true if `duration = false`, because then profiles would work similarly to Prometheus metrics.

However, if `duration = true`, wouldn't a lower scrape frequency lead to less catched events?

E.g. if `scrape_frequency = "15s"`, then each observation will last 14 seconds. For every 60 seconds we will lose 4 seconds of observations - this is the time between scrapes. On the other hand, for `scrape_frequency = "60s"` we will only lose 1 second worth of samples for every minute.

It'd be great if someone from the Pyroscope team confirms this.

Reply:
The problem is when a bug leads to e.g. a rapid growth in memory and within 20 seconds the instance OOMs. This is much harder to catch with 1m scrape interval, as you'd need to be lucky to finish the profile just a few seconds before OOM event happens.

Reply:
I agree that when analysing OOMs, it's better to have a short scrape frequency. However, in case the application is not crashing and `duration = true`, I think that longer scrapes will actually result in more samples and will better detect rapid changes.

Reply:
What do you refer to in `duration = false|true`? I don't see such setting on pyroscope.scrape.

Reply:
Sorry, I meant `delta = true`. I'll edit my comment above.